testing/integration_test.go

```diff
@@ -0,0 +1,235 @@
+	runID  = time.Now().Unix()
+)
+
+const cloudScope = "https://www.googleapis.com/auth/cloud-platform"
+
+const startupTemplate = `
+#! /bin/bash
+
+# Fail on any error
+set -eo pipefail
+
+# Display commands being run
+set -x
+# Install git
+sudo apt-get update
+sudo apt-get -y -q install git-all build-essential
+
+# Install desired version of Node.js
+curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.8/install.sh | bash
+export NVM_DIR="$HOME/.nvm"
+[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh" # This loads nvm
+nvm install {{.NodeVersion}}
+npm -v
+node -v
+
+# Install agent
+git clone https://github.com/GoogleCloudPlatform/cloud-profiler-nodejs.git
+cd cloud-profiler-nodejs
+git pull {{.Repo}} {{.Branch}}
+git reset --hard {{.Commit}}
+npm install
+npm run compile
+npm pack
+VERSION=$(node -e "console.log(require('./package.json').version);")
+PROFILER="$HOME/cloud-profiler-nodejs/google-cloud-profiler-$VERSION.tgz"
+
+TESTDIR="$HOME/test"
+mkdir -p "$TESTDIR"
+cp "testing/busybench.js" "$TESTDIR"
+cd "$TESTDIR"
+
+npm install "$PROFILER"
+
+# Run benchmark with agent
+GCLOUD_PROFILER_LOGLEVEL=5 GAE_SERVICE={{.Service}} node --require @google-cloud/profiler busybench.js 600
+
+# Indicate to test that script has finished running
+echo "busybench finished profiling"
+`
+
+type nodeGCETestCase struct {
+	proftest.GCETestConfig
+	name             string
+	nodeVersion      string
+	wantProfileTypes []string
+}
+
+func (inst *nodeGCETestCase) initializeStartUpScript(template *template.Template) error {
+	var buf bytes.Buffer
+	err := template.Execute(&buf,
+		struct {
+			Service     string
+			NodeVersion string
+			Repo        string
+			Branch      string
+			Commit      string
+		}{
+			Service:     inst.service,
+			NodeVersion: inst.nodeVersion,
+			Repo:        *repo,
+			Branch:      *branch,
+			Commit:      *commit,
+		})
+	if err != nil {
+		return fmt.Errorf("failed to render startup script for %s: %v", inst.Name, err)
+	}
+	inst.StartupScript = buf.String()
+	return nil
+}
+
+func TestAgentIntegration(t *testing.T) {
+	projectID := os.Getenv("GCLOUD_TESTS_NODEJS_PROJECT_ID")
+	if projectID == "" {
+		t.Fatalf("Getenv(GCLOUD_TESTS_NODEJS_PROJECT_ID) got empty string")
+	}
+
+	zone := os.Getenv("GCLOUD_TESTS_NODEJS_ZONE")
+	if zone == "" {
+		t.Fatalf("Getenv(GCLOUD_TESTS_NODEJS_ZONE) got empty string")
+	}
+
+	if *commit == "" {
+		I12aedd58f181f365849e0ee3db28daf307075129
+		t.Fatal("commit flag is not set")
+	}
+
+	ctx := context.Background()
+
+	client, err := google.DefaultClient(ctx, cloudScope)
+	if err != nil {
+		t.Fatalf("failed to get default client: %v", err)
+	}
+
+	computeService, err := compute.New(client)
+	if err != nil {
+		t.Fatalf("failed to initialize compute Service: %v", err)
+	}
+
+	template, err := template.New("startupScript").Parse(startupTemplate)
+	if err != nil {
+		t.Fatalf("failed to parse startup script template: %v", err)
+	}
+
+	tr := proftest.TestRunner{
```

Comment:
Can you get rid of the tr variable by simply inlining the initialization below like
```
gceRunner := proftest.GCETestRunner{
  TestRunner: {Client: client},
  ComputeService: computeService,
}
```
?

Reply:
Done.